osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -72,6 +74,39 @@ private void load([CanBeNull] SkinEditorOverlay skinEditor)
                 new ExportSkinButton(),
                 new DeleteSkinButton(),
             };
+            bool cancel = false;
+
+            held.BindValueChanged((changeEvent) =>
+            {
+                // We're doing this to cancel the next event if we keep the changes
+                // If we don't do this it'll get in an infinite bindable loop
+                if (cancel)
+                {
+                    cancel = false;
+
+                    return;
+                }
+
+                SkinEditor.SkinEditor editor = skinEditor.SkinEditor;
+
+                // We just test if the editor isnt null and the user has mutated something, if they have we can
+                // proceed and request the change dialog
+                if (editor != null && editor.Mutated)
+                {
+                    editor.RequestChange(update, () =>
+                    {
+                        cancel = true;
+
+                        held.Value = changeEvent.OldValue;
+                    });
+
+                    return;
+                }
+
+                update();
+            }, true);
```

Comment:
The enforced connection here between the settings overlay and the skin editor is definitely a no-no. This should exist inside `SkinEditor`, before it proceeds with updating its state to the new skin.